pkg/action/scan.go

```diff
@@ -118,22 +103,27 @@ func scanSinglePath(ctx context.Context, c malcontent.Config, path string, ruleF
 		return nil, err
 	}
 	fd := f.Fd()
-	defer f.Close()
 
 	fi, err := f.Stat()
 	if err != nil {
 		return nil, err
 	}
 
 	size := fi.Size()
-	if size == 0 {
+	if size <= 0 {
```

Comment:
why can size be less than zero?

Reply:
Updated in [`95e359c` (#1078)](https://github.com/chainguard-dev/malcontent/pull/1078/commits/95e359c9b9aa9272519d09651b36db8b2b586e26).